File: Python/lib/Calculator.py
from msvcrt import getch
from os import system


def add():
    var1 = float(input('Enter the Number: '))

    while 1:
        var2 = float(input('Enter the Number: '))
        if var2 == 0:
            break
        var1 = var1 + var2
    print('Result = ', var1)
    system('pause')


def sub():
    var1 = float(input('Enter the Number: '))

    while 1:
        var2 = float(input('Enter the Number: '))
        if var2 == 0:
            break
        var1 = var1 - var2
    print('Result = ', var1)
    system('pause')

def multi():
    var1 = float(input('Enter the Number: '))

    while 1:
        var2 = float(input('Enter the Number: '))
        if var2 == 0:
            break
        var1 = var1 * var2
    print('Result = ', var1)
    system('pause')

def div():
    var1 = float(input('Enter the Number: '))

    while 1:
        var2 = float(input('Enter the Number: '))
        if var2 == 0:
            break
        var1 = var1 / var2
    print('Result = ', var1)
    system('pause')

def Calculator():
    while 1:
        system('cls')
        print('0. Exit')
        print('1. Add')
        print('2. Sub')
        print('3. Multi')
        print('4. Div\n')

        choice = int(input('Enter your Choice: '))

        if choice == 1:
            print('')
            add()
        elif choice == 2:
            print('')
            sub()
        elif choice == 3:
            print('')
            multi()
        elif choice == 4:
            print('')
            div()
        elif choice == 0:
            quit()
        else:
            print('')
            print('Error!')
            getch()
            system('cls')
        print('')

Calculator()

File: Java/src/lib/Calculator.java
package lib;

import java.util.Scanner;

public class Calculator {
	
	double var1;
	double var2;

	Scanner input = new Scanner(System.in);

	/***
	 * Md Rashed Karim
	 * This is add method
	 * @return sum
	 */
	public double Add() {
		
		double var1;
		double var2;
		double sum = 0;
		
		System.out.print("Enter the Number: ");
		var1 = input.nextDouble();
		
		do{	
			System.out.print("Enter the Number: ");
			var2 = input.nextDouble();
			
			if (var2 == 0) {
				break;
			}	
			
			sum = (var1 = var1 + var2); //var1+=var2
			
		}while(!(var2 == 0));
			
		return sum;
	}

	/***
	 * Tahsin
	 * @return sum
	 */
	public double Sub() {
		
		double var1;
		double var2;
		double sum = 0;
		
		System.out.print("Enter the Number: ");
		var1 = input.nextDouble();
		
		do{	
			System.out.print("Enter the Number: ");
			var2 = input.nextDouble();
			
			if (var2 == 0) {
				break;
			}	
			
			sum = (var1 = var1 - var2); //var1+=var2
			
		}while(!(var2 == 0));
			
		return sum;
	}
	
	public double Multiplication() {
		
		double var1;
		double var2;
		double sum = 0;
		
		System.out.print("Enter the Number: ");
		var1 = input.nextDouble();
		
		do{	
			System.out.print("Enter the Number: ");
			var2 = input.nextDouble();
			
			if (var2 == 0) {
				break;
			}	
			
			sum = (var1 = var1 * var2); //var1+=var2
			
		}while(!(var2 == 0));
			
		return sum;
	}
	
	public double Div() {
		
		double var1;
		double var2;
		double sum = 0;
		
		System.out.print("Enter the Number: ");
		var1 = input.nextDouble();
		
		do{	
			System.out.print("Enter the Number: ");
			var2 = input.nextDouble();
			
			if (var2 == 0) {
				break;
			}	
			
			sum = (var1 = var1 / var2); //var1+=var2
			
		}while(!(var2 == 0));
			
		return sum;
	}
	public Calculator() {
		int c;
		do {			
			System.out.println("1. Addition");
			System.out.println("2. Subtraction");
			System.out.println("3. Multiplication");
			System.out.printf("4. Division\n");
			System.out.println("0. Exit\n");
			
			System.out.print("Enter choice: ");
			
			c = input.nextInt();
			
			if (c == 1) {
				System.out.println(this.Add() + "\n");	
			}else if (c == 2) {			
				System.out.println(this.Sub() + "\n");
			}else if (c == 3) {			
				System.out.println(this.Multiplication() + "\n");
			}else if (c == 4) {
				System.out.println(this.Div() + "\n");			
			}else if (c == 0) {
				break;
			}else {
				System.out.println("Error");
			}	
		}while(!(c == 0));
	}
}
